mempool/src/shared_mempool.rs

```diff
@@ -36,10 +60,38 @@ use vm_validator::vm_validator::{get_account_state, TransactionValidation};
 struct PeerSyncState {
     timeline_id: u64,
     is_alive: bool,
+    network_id: usize,
 }
 
+/// stores only peers that receive txns from this node
 type PeerInfo = HashMap<PeerId, PeerSyncState>;
 
+/// Handle in shared mempool to network-related components
+pub struct MempoolNetworkHandle {
+    /// network sender for this network
+    pub network_sender: MempoolNetworkSender,
+    /// events from the network to this node
+    pub network_events: MempoolNetworkEvents, // events come to this from upstream node to this node
+    /// true if the peers to this node in this network are upstream peers or not
+    pub is_validator_network: bool, // true if this peer is from upstream network or from validator network
+}
+
+#[derive(Clone)]
+struct MempoolNetworkManager {
+    network_senders: Vec<MempoolNetworkSender>,
+    upstream_peers: Vec<PeerId>,
```

Comment:
Do upstream peers also include full nodes peers in the same network?

Reply:
that should be what's happening, since we're getting the upstream peers from the state sync config